MediaGallery/MediaGallery/SaveMedia.ios.cs

```diff
@@ -36,26 +37,80 @@ static async Task PlatformSaveAsync(MediaFileType type, Stream fileStream, strin
                 await fileStream.CopyToAsync(stream);
                 stream.Close();
 
-                await PlatformSaveAsync(type, filePath).ConfigureAwait(false);
+                await PlatformSaveAsync(type, filePath, albumName).ConfigureAwait(false);
             }
             finally
             {
                 DeleteFile(filePath);
             }
         }
 
-        static async Task PlatformSaveAsync(MediaFileType type, string filePath)
+        static async Task PlatformSaveAsync(MediaFileType type, string filePath, string albumName = null)
         {
             using var fileUri = new NSUrl(filePath);
+            
+            PHAssetCollection collection = null;
+            // If albumName is null we do what we always used to do and not create an album.
+            // If albumName is an empty string we don't wish to create an album (which is the same as null for iOS)
+            // Otherwise we wish to create an album.
+            if (String.IsNullOrEmpty(albumName) == false)
+            {
+                // Fetch album.
+                var fetchOptions = new PHFetchOptions()
+                {
+                    Predicate = NSPredicate.FromFormat($"title=\"{albumName}\"")
+                };
+                collection = PHAssetCollection.FetchAssetCollections(PHAssetCollectionType.Album, PHAssetCollectionSubtype.AlbumRegular, fetchOptions).FirstObject as PHAssetCollection;
```

Comment:
Should we request additional permissions? On which versions of iOS have you tested this?

Reply:
As far as I am aware no additional permissions are required to make an album. Our production app has only had `NSPhotoLibraryUsageDescription` in it for about 7 years. 

It was only until we moved to using Xamarin.MediaGallery did we even realise there was a new `NSPhotoLibraryAddUsageDescription` permission.

I have only been testing the new code on iOS 16.5, but the source where I originally wrote would have been at least iOS 10 and up.

Reply:
If I specify the album name, it asks for additional permission. I don't really like this behavior. It will not be clear to users. Why should the user be able to select the photos to which he provides access when saving the photo?


https://github.com/dimonovdd/Xamarin.MediaGallery/assets/59065470/a2992156-403a-453d-a76e-a27abff22401